content/How to Test React Native Apps - Manuela Viola.md

```diff
@@ -0,0 +1,143 @@
+
+## About @testing-library/react-native
+
+One important thing to point out is that this testing library has changed many times and with migrations needed from one version to another. This tutorial will work with the latest version at the time of writing it (`7.0.1`). For migrations you can check their [migration guide](https://callstack.github.io/react-native-testing-library/docs/migration-v7/#guide-for-testing-libraryreact-native-users)
+
+## App to be tested
+
+For this tutorial, we are going to use [this](https://github.com/ManuViola77/testing_react_native_apps) simple app, but of course you can apply everything learned in any app you want. In this app we just have two simple screens with some components we can test.
+
+## Configuration
+
+Since we are going to use the `@testing-library/react-native` library, we need to install it. To use the specific `7.0.1` version we can install it like this:
+`npm install @testing-library/react-native@7.0.1`
+
+Also, we need to configure jest. We should create a `jest.config.js` file in the projects root folder and for our example we can fill it with this content:
+
+https://gist.github.com/ManuViola77/d98f577c8ea35655d96dcd4654ab398e
+
+I will mention some of these configurations later in this tutorial and [here](https://jestjs.io/docs/en/configuration) you can find more information.
+
+## Tests folder
+
+As you can see, our project already has a `__tests__` folder with an `App-test.js` file inside. This tests if the app is created and rendered correctly.
+
+## Creating tests
+
+To actually create a test, we should create a file with a `.spec.js` extension inside the `__tests__` folder. We could also have folders inside for organizing the tests, for example, one folder called `screens` and have inside one test per screen.
+
+Inside these tests, we are going to use `describe()` just to describe what we want to test, `it()` (same as `test()`) to create the tests we want to have and `expect()` to create our expected behaviour. Also, we will be using `beforeEach()` to execute some code before each `it()` test. [Here](https://jestjs.io/docs/en/api) is a complete list of `jest` methods to use inside tests.
+
+Also, we use the `@testing-library/react-native` library to test our components behaviour. We could use `fireEvent` to fire a button pressed event or `waitFor()` to wait for promises results. [Here](https://www.native-testing-library.com/docs/cheat-sheet) is a cheat sheet with the possible methods to use.
+
+Notice that to know how to ask for a specific component, we can simply add the `testID="componentId"` property to the component we want to identify. To simplify, I already added this property to all the components we are testing.
+
+## Creating tests for our App
+
+Now we need to create our own tests, so we can test our `MainScreen` and `AnotherScreen`. We can create the `screens` folder inside the `__tests__` folder to have better organized tests.
```

Comment:
I haven't seen your example app yet, but maybe `AnotherScreen` is not the best naming? could you please use a more descriptive name?